research/fed-sm/job_configs/fedsm_prostate/app/custom/learners/supervised_learner.py

```diff
@@ -0,0 +1,311 @@
+# Copyright (c) 2021-2022, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from abc import abstractmethod
+
+import numpy as np
+import torch
+from torch.utils.tensorboard import SummaryWriter
+
+from nvflare.apis.dxo import DXO, DataKind, MetaKey, from_shareable
+from nvflare.apis.fl_constant import FLContextKey, ReturnCode
+from nvflare.apis.fl_context import FLContext
+from nvflare.apis.shareable import Shareable, make_reply
+from nvflare.apis.signal import Signal
+from nvflare.app_common.abstract.learner_spec import Learner
+from nvflare.app_common.app_constant import AppConstants, ValidateType
+
+
+class SupervisedLearner(Learner):
```

Comment:
yeah, currently it is reusing the learner here https://github.com/NVIDIA/NVFlare/tree/dev/examples/advanced/prostate/prostate_2D/custom/learners
but agree, since these are not part of core, there is no need to used them forcefully.
Let's discuss how to reasonably refactor